plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -520,11 +524,153 @@ int add_event(struct flb_cloudwatch *ctx, struct cw_flush *buf,
     return 0;
 }
 
+int should_add_to_emf(struct flb_ir_metric *an_item)
+{
+    // Valid for cpu plugin
+    if (strncmp(an_item->key.via.str.ptr, "cpu_p", 4) == 0 || strncmp(an_item->key.via.str.ptr, "user_p", 6) == 0 || strncmp(an_item->key.via.str.ptr, "system_p", 8) == 0)
+    {
+        return 1;
+    }
+
+    // Valid for mem plugin
+    if (strncmp(an_item->key.via.str.ptr, "Mem.total", 9) == 0 || strncmp(an_item->key.via.str.ptr, "Mem.used", 8) == 0 || strncmp(an_item->key.via.str.ptr, "Mem.free", 8) == 0 
+    || strncmp(an_item->key.via.str.ptr, "Swap.total", 10) == 0 || strncmp(an_item->key.via.str.ptr, "Swap.used", 9) == 0 || strncmp(an_item->key.via.str.ptr, "Swap.free", 9) == 0)
+    {
+        return 1;
+    }
+
+    return 0;
+}
+
+struct msgpack_object pack_emf_payload(struct flb_cloudwatch *ctx, struct mk_list *flb_ir_metrics, const char *input_plugin, struct flb_time tms)
+{
+    struct mk_list *metric_temp;
+    struct mk_list *metric_head;
+
+    /* msgpack::sbuffer is a simple buffer implementation. */
+    msgpack_sbuffer sbuf_emf;
+    msgpack_sbuffer_init(&sbuf_emf);
+
+    /* Serialize values into the buffer using msgpack_sbuffer_write callback function. */
+    msgpack_packer packer_emf;
+    msgpack_packer_init(&packer_emf, &sbuf_emf, msgpack_sbuffer_write);
+    msgpack_pack_map(&packer_emf, (mk_list_size(flb_ir_metrics)) + 1);
+
+    //Pack the _aws map
+    msgpack_pack_str(&packer_emf, 4);
+    msgpack_pack_str_body(&packer_emf, "_aws", 4);
+
+    msgpack_pack_map(&packer_emf, 2);
+
+    msgpack_pack_str(&packer_emf, 9);
+    msgpack_pack_str_body(&packer_emf, "Timestamp", 9);
+    msgpack_pack_long_long(&packer_emf, tms.tm.tv_sec * 1000L);
+
+    msgpack_pack_str(&packer_emf, 17);
+    msgpack_pack_str_body(&packer_emf, "CloudWatchMetrics", 17);
+    msgpack_pack_array(&packer_emf, 1);
+
+    msgpack_pack_map(&packer_emf, 3);
+
+    msgpack_pack_str(&packer_emf, 9);
+    msgpack_pack_str_body(&packer_emf, "Namespace", 9);
+
+    if (ctx->metric_namespace)
+    {
+        msgpack_pack_str(&packer_emf, flb_sds_len(ctx->metric_namespace));
+        msgpack_pack_str_body(&packer_emf, ctx->metric_namespace, flb_sds_len(ctx->metric_namespace));
+    }
+    else
+    {
+        msgpack_pack_str(&packer_emf, 18);
+        msgpack_pack_str_body(&packer_emf, "fluent-bit-metrics", 18);
+    }
+
+    msgpack_pack_str(&packer_emf, 10);
+    msgpack_pack_str_body(&packer_emf, "Dimensions", 10);
+
+    struct mk_list *head;
+    struct flb_split_entry *dimension_list;
+    if (ctx->metric_dimensions)
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions